src/include/qmgr.h

```diff
@@ -79,6 +79,11 @@
                             pstderr("qmgr: Out of memory\n"); \
                             clean_up_and_exit(5); \
                         }
+/* This macro will duplicate string */
+#define Mstrdup(x,y)    if ( (x=strdup(y)) == NULL ) { \
```

Comment:
I am usually () happy when I write macros.  You don't know what is going to be passed into the macro.  It can be something simple as foo, or foo + 5, or something even more complicated.  You are dependent on the order of operations of operators unless you use extra ().